nitrates/analysis_seeds/UpperLimits.py

```diff
@@ -0,0 +1,362 @@
+        rate_upper_limit = 5*rate_std
+       # ul_5sigma_new = []
+
+   # Using 4 spectral templates
+        alpha_values = [-1.0,-1.9,0.62,-1.5]
+        beta_values = [-2.3,-3.7,0.0,0.0]
+        Epeak_values = [230.0,70.0,185.0,1500.0]
+
+        for index,j in enumerate(alpha_values):
+        #for j in range(1):
+                      
+            if index == 0:
+                spec_temp = "Band_Normal_template_Ep230keV_alpha-1.0_beta-2.3"
+            elif index == 1:
+                spec_temp = "Band_Soft_template_Ep70keV_alpha-1.9_beta_-3.7"
+            elif index == 2:
+                spec_temp = "Cutoff_PL_GW170817-like_Ep185keV_alpha0.62"
+            elif index == 3:
+                spec_temp = "Cutoff_PL_hard_Ep1500_keV_alpha-1.5"
+            
+            ul_5sigma_new = []
+
+            for i in range(len(theta_values)):
+           # for i in range(3):
+                drm_tab_new = get_resp4ul_tab(theta_values[i],phi_values[i])
+
+# using energy bin 15-350 and ignoring 350-500
+                chan_low = 0
+                chan_hi = 3
+# response matrix using selected energy bins and corrected for number of active dets
+                drm_matrix_new = drm_tab_new['MATRIX'][:,chan_low:(chan_hi+1)] * Ndet_ratio
+
+# find the flux that gives an expected rate equal to the rate upper limit
+                if index>=2:
+                    flux_upper_limit_new = rate2comp_eflux(rate_upper_limit,drm_matrix_new,\
+                                               drm_tab_new['ENERG_LO'], drm_tab_new['ENERG_HI'],\
+                                               alpha_values[index],Epeak_values[index], flux_elo, flux_ehi)
+                else:
+                    flux_upper_limit_new = rate2band_eflux(rate_upper_limit, drm_matrix_new,\
+                                                   drm_tab_new['ENERG_LO'], drm_tab_new['ENERG_HI'],\
+                                                   alpha_values[index], beta_values[index], Epeak_values[index], flux_elo, flux_ehi)
+                
+                ul_5sigma_new.append(flux_upper_limit_new)
+            
+            filename = f"ul_5sigma_{spec_temp}_{dur_values[k]}.txt"
+            with open(filename, "w") as file:
+                #for item in ul_5sigma_new:
+                    #file.write(str(item) + "\n")
+                for i in range(len(ul_5sigma_new)):
+                    file.write(f"{ul_5sigma_new[i]}\t{theta_values[i]}\t{phi_values[i]}\t{ras[i]}\t{decs[i]}\n")    
+            logging.debug('5 sigma UL values for:')
+            logging.debug(spec_temp)
+            logging.debug('and time bin')
+            logging.debug(dur_values[k])
+            logging.debug('-----')
+            logging.debug(ul_5sigma_new)
+            logging.debug('------')
+
+
+    elapsed_time = time.time() - start_time
+
+# Print the elapsed time
+    print(f"Time taken: {elapsed_time} seconds")
+
+# ------------------------------------
+# Old using grid_ids for infov----
```

Comment:
remove commented code if its not going to be used

Reply:
The commented old portion of the code is now removed